### DISCUSSION TITLE
How to wrap javascript in vue.js?

### DISCUSSION THREAD

**Discussion by toshikijp** [andatoshiki/toshiki-discuss#16] [Q&A] [answered]
As the question itself describes

**Comment by andatoshiki** [comment, marked as answer]
If you really want a simple and straightforward solution, you can simply put all of that within the `mounted` callback: that is when the component template has been parsed and the DOM constructed, so `#countdown` should be accessible by then. [[Read more about lifecycle hooks in VueJS](https://vuejs.org/v2/guide/instance.html#Instance-Lifecycle-Hooks)](https://vuejs.org/v2/guide/instance.html#Instance-Lifecycle-Hooks).

Note: However, if you want to really learn how to use VueJS properly, please continue reading on how to create a custom timer component at the end of this answer :) by stuffing everything into the `mounted` callback, you are missing out the fun part of authoring your first and rather simple VueJS component.

Before we get started, note that there is a mistake when you invoke `timer()` in your `setInterval` callback. It should be:

```js
var countdownTimer = setInterval(timer, 1000);
```

Here is a proof-of-concept example, where you "turkey stuff" all your timer logic into the `mounted()` callback:



```js
new Vue({
  el: '#app',
  template: '#custom-component',
  mounted: function() {
    var upgradeTime = 7200;
    var seconds = upgradeTime;

    function timer() {
      var days = Math.floor(seconds / 24 / 60 / 60);
      var hoursLeft = Math.floor((seconds) - (days * 86400));
      var hours = Math.floor(hoursLeft / 3600);
      var minutesLeft = Math.floor((hoursLeft) - (hours * 3600));
      var minutes = Math.floor(minutesLeft / 60);
      var remainingSeconds = seconds % 60;
      if (remainingSeconds < 10) {
        remainingSeconds = "0" + remainingSeconds;
      }
      document.getElementById('countdown').innerHTML = hours + ":" + minutes + ":" + remainingSeconds;
      if (seconds == 0) {
        clearInterval(countdownTimer);
        document.getElementById('countdown').innerHTML = "Completed";
      } else {
        seconds--;
      }
    }
    var countdownTimer = setInterval(timer, 1000);
  }
})
```

```html
<script src="https://cdnjs.cloudflare.com/ajax/libs/vue/2.5.13/vue.min.js"></script>
<div id="app"></div>

<script type="text/x-template" id="custom-component">
  <div>
    <p>I am a custom component.</p>
    <p>
      My timer:<br />
      <span id="countdown" class="timer"></span>
    </p>
  </div>
</script>
```
